### PR TITLE
Improve extra-langs and move Racket support to its own layer

### PR DESCRIPTION
**Pull request by trishume**
Fixes some minor things in minor languages:
- Lifetime specifiers in Rust no longer pair
- `company-quickhelp-mode` is now disabled in `racket-mode`. It had a bad interaction where it would open full doc windows every time the autocomplete popup appeared.
- Adds support for the [Nim](http://nim-lang.org/) language.

And also moves Racket to its own mode since its config set had outgrown the `extra-langs` layer.

### CODE REVIEW

File: contrib/lang/racket/packages.el [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        "ml" 'evil-lisp-state
+        "mt" 'racket-test
+        "mr" 'racket-run
+        "mg" 'racket-visit-definition
+        "mhd" 'racket-doc)
+      ;; Bug exists in Racket company backend that opens docs in new window when
+      ;; company-quickhelp calls it. Note hook is appendended for proper ordering.
+      (when (configuration-layer/package-declaredp 'company-quickhelp)
+        (add-hook 'company-mode-hook
+                  '(lambda () (when (equal major-mode 'racket-mode) (company-quickhelp-mode -1))) t))
```
Comment (syl20bnr) [inline contrib/lang/racket/packages.el]: Does this alternative works for you ?
```elisp
(when (configuration-layer/package-declaredp 'company-quickhelp)
        (add-hook 'racket-mode-hook '(lambda () (company-quickhelp-mode -1))))
```

File: contrib/lang/racket/packages.el [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      (add-hook 'racket-mode-hook
+                '(lambda ()
+                   (define-key racket-mode-map (kbd "H-r") 'racket-run))))))
```
Comment (syl20bnr) [inline contrib/lang/racket/packages.el]: You should be able to define this key outside of a hook since you are in `:config`.

**Comment by syl20bnr** [comment]
Great PR !
The logo for racket is too big, the recommended height is 200px.

File: contrib/lang/racket/packages.el (hunk 1 above)
Comment (trishume) [inline contrib/lang/racket/packages.el]: That was the first thing I tried, `company-mode-hook` is called after `racket-mode-hook` and `company-quickhelp-mode` is activated in the company hook so it got reactivated right away.

File: contrib/lang/racket/packages.el (hunk 2 above)
Comment (trishume) [inline contrib/lang/racket/packages.el]: Derp, didn't realize the redundancy. I wrote that a long time ago when I was still in the "herp derp copy other layers" phase and obviously I copied two redundant patterns.

**Comment by trishume** [comment]
Fixed.

**Comment by syl20bnr** [comment]
Great ! I love the new racker layer :+1: 
I applied the [conventions](https://github.com/syl20bnr/spacemacs/blob/master/doc/CONVENTIONS.md) to racket layer and added more key bindings, see commit 3e68321f6d44c466bf966f7e202dbdc1aa62c1a6

Thank you !
Cherry-picked in `develop`, you can safely delete your branch.

**Comment by trishume** [comment]
Whoah cool, thanks for the improvements!

**Pull request closed without merging**